packages/cdk-assets/lib/private/handlers/container-images.ts

```diff
@@ -29,7 +31,7 @@ export class ContainerImageAssetHandler implements IAssetHandler {
       throw new Error(`No ECR repository named '${destination.repositoryName}' in account ${account}. Is this account bootstrapped?`);
     }
 
-    const imageUri = `${repoUri}:${destination.imageTag}`;
+    const imageUri = isExternalSource(this.asset.source) ? await shell(this.asset.source.executable.split(' ')) : `${repoUri}:${destination.imageTag}`;
```

Comment:
Yeah naw, I know this works but it's conflating responsibilities.

I'd rather you take 2 completely different paths through the code based on the asset type (2 different private methods, probably).

The "don't Docker build if already exists" check is only for the ONE branch (the self-build branch) -- in case of the external build we expect the external tool to do its own build deduplication.